docroot/modules/custom/civictheme_migrate/tests/src/Kernel/NodeHelperTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Drupal\Tests\civictheme_migrate\Kernel\Utils;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Drupal\civictheme_migrate\Utils\NodeHelper;
+use Drupal\node\Entity\NodeType;
+use Drupal\Core\Url;
+
+/**
+ * Tests the NodeHelper class.
+ *
+ * @coversDefaultClass \Drupal\civictheme_migrate\Utils\NodeHelper
+ *
+ * @group civictheme_migrate
+ * @group site:kernel
+ */
+class NodeHelperTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'civictheme_migrate',
+    'file',
+    'media',
+    'node',
+    'migrate',
+    'user',
+    'system',
+    'image',
+    'field',
+    'text',
+    'path',
+    'path_alias',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('file');
+    $this->installEntitySchema('path_alias');
+    $this->installSchema('system', ['sequences']);
+    $this->installSchema('user', ['users_data']);
+    $this->installSchema('node', ['node_access']);
+    $this->installSchema('file', 'file_usage');
+    $this->installConfig(['file', 'image', 'media', 'node', 'system', 'field']);
+
+    // Create the custom node type.
+    $type = NodeType::create([
+      'type' => 'type1',
+      'name' => 'Custom Type 1',
+      'description' => 'A custom node type.',
+      'display_submitted' => FALSE,
+    ]);
+    $type->save();
+
+    // Create the custom node type.
+    $type = NodeType::create([
+      'type' => 'type2',
+      'name' => 'Custom Type 2',
+      'description' => 'A custom node type.',
+      'display_submitted' => FALSE,
+    ]);
+    $type->save();
+
+    // Create and save a node with ID '1'.
+    $node = Node::create([
+      'type' => 'type1',
+      'nid' => 99999999,
+      'title' => 'Title 1',
+      'status' => 1,
+      'path' => [
+        'alias' => '/alias1',
+      ],
+    ]);
+    $node->save();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function tearDown(): void {
+    // Clean up the created node with ID '1'.
+    $node = Node::load(99999999);
+    if ($node instanceof NodeInterface) {
+      $node->delete();
+    }
+
+    parent::tearDown();
+  }
+
+  /**
+   * Tests the lookupNodeFromAlias method.
+   *
+   * @dataProvider nodeAliasDataProvider
+   *
+   * @covers ::lookupNodeFromAlias
+   */
+  public function testLookupNodeFromAlias(string $alias, ?string $expectedNodeId) {
+    // Mock the Url class and provide the necessary route parameters.
+    $routeParameters = ['node' => (int) $expectedNodeId];
+    $url = $this->createMock(Url::class);
+    $url->method('getRouteParameters')
```

Comment:
The getRouteParameters method relies on the Drupal routing system, which is not fully available in a kernel test environment.